tests/helpers/test_httpx_client.py

```diff
@@ -0,0 +1,143 @@
+"""Test the httpx client helper."""
+
+import httpx
+import pytest
+
+from homeassistant.core import EVENT_HOMEASSISTANT_CLOSE
+import homeassistant.helpers.httpx_client as client
+
+from tests.async_mock import Mock, patch
+
+
+async def test_async_get_async_client_with_ssl(hass):
+    """Test init async client with ssl."""
+    client.async_get_async_client(hass)
+
+    assert isinstance(hass.data[client.DATA_ASYNC_CLIENT], httpx.AsyncClient)
+
+
+async def test_async_get_async_client_without_ssl(hass):
+    """Test init async client without ssl."""
+    client.async_get_async_client(hass, verify_ssl=False)
+
+    assert isinstance(hass.data[client.DATA_ASYNC_CLIENT_NOVERIFY], httpx.AsyncClient)
+
+
+async def test__async_create_async_httpx_client_with_ssl_and_cookies(hass):
+    """Test init async client with ssl and cookies."""
+    client.async_get_async_client(hass)
+
+    httpx_client = client._async_create_async_httpx_client(hass, cookies={"bla": True})
```

Comment:
If we're going to allow creating a client with other options, we need to make the interface non private.

Reply:
👍  done